### PR TITLE
Setup Dutch Scenario & Liquidation Fixes

### PR DESCRIPTION
**Pull request by arturbeg**

### CODE REVIEW

File: core/src/libraries/account/AccountChecks.sol [hunk 1]
```diff
@@ -100,17 +102,13 @@ library AccountChecks {
         returns (MarginInfo memory marginInfo)
     {
         marginInfo = self.getMarginInfoByBubble(token);
-        bool isBelowAdl = marginInfo.adlDelta < 0;
+        bool isBelowAdl = marginInfo.adlDelta < 0 + self.getPendingAutoExchangeFunds(token).toInt();
```
Comment (0xZenus) [inline core/src/libraries/account/AccountChecks.sol:105]: is it expected to add the pending funds to zero?

File: core/src/libraries/account/AccountChecks.sol [hunk 2]
```diff
@@ -128,7 +126,13 @@ library AccountChecks {
         }
     }
 
-    function checkBetweenGlobalAdlAndLm(Account.Data storage self) internal view returns (MarginInfo memory) {
-        return self.checkBetweenLocalAdlAndLm(address(0));
+    function checkBelowGlobalLm(Account.Data storage self) internal view returns (MarginInfo memory marginInfo) {
+        marginInfo = self.checkAboveLocalAdl(address(0));
```
Comment (0xZenus) [inline core/src/libraries/account/AccountChecks.sol:130]: this checks if the account is globally above ADL and below LM, so why is the function name changed?

File: core/src/libraries/account/AccountLiquidation.sol [hunk 3]
```diff
@@ -558,7 +558,7 @@ library AccountLiquidation {
             self.liquidationBidPriorityQueuesPerBubble[quoteToken];
 
         // revert if the account is above dutch margin requirement & the liquidation bid queue is not empty
-        bool isAboveDutch = (marginInfo.dutchDelta > 0) && (localMarginInfo.dutchDelta > 0);
+        bool isAboveDutch = localMarginInfo.dutchDelta + self.getPendingAutoExchangeFunds(quoteToken).toInt() > 0;
```
Comment (0xZenus) [inline core/src/libraries/account/AccountLiquidation.sol:561]: For efficiency, I'd do the following:
-> if liquidation bid is empty => check if the account is locally between ADL and LM
-> if liquidation is not empty => check if the account is locally between ADL and Dutch (Also please wrap this up in a checker in AccountChecks)

This way, we do only one local check per branch (in comparison to here where we do two checks).

File: core/src/libraries/account/AccountLiquidation.sol [hunk 4]
```diff
@@ -558,7 +558,7 @@ library AccountLiquidation {
...
 
         if (
             !(liquidationBidPriorityQueues.latestQueueEndTimestamp == 0)
```
Comment (0xZenus) [inline core/src/libraries/account/AccountLiquidation.sol:564]: Could we have a helper function in liquidation bid queue library called "isEmpty" ?

File: core/src/libraries/account/AccountChecks.sol (hunk 1 above)
Comment (arturbeg) [reply core/src/libraries/account/AccountChecks.sol:105]: oh, good catch, it should be added to adlDelta..

Comment (arturbeg) [reply core/src/libraries/account/AccountChecks.sol]: https://github.com/Voltz-Protocol/v2-core/pull/228/commits/2fff05ce2013f6a9c8c5e70aa9d4bc5425506bcd

File: core/src/libraries/account/AccountChecks.sol (hunk 2 above)
Comment (arturbeg) [reply core/src/libraries/account/AccountChecks.sol]: another good catch
https://github.com/Voltz-Protocol/v2-core/pull/228/commits/d17d627149787a12dbe51d42c4bb8d2c0e4bfd47

File: core/src/libraries/account/AccountLiquidation.sol (hunk 3 above)
Comment (arturbeg) [reply core/src/libraries/account/AccountLiquidation.sol]: https://github.com/Voltz-Protocol/v2-core/pull/228/commits/01e4489c8345ea88faac1024eadcddb716b15940

File: core/src/libraries/account/AccountLiquidation.sol (hunk 4 above)
Comment (arturbeg) [reply core/src/libraries/account/AccountLiquidation.sol]: https://github.com/Voltz-Protocol/v2-core/pull/228/commits/953d0498d31ffe997c3aba8768409d7eb6210627

Comment (arturbeg) [reply core/src/libraries/account/AccountLiquidation.sol]: can perhaps add further more granular helpers in other prs that also cover varaitions fo this check

File: core/src/libraries/LiquidationBidPriorityQueue.sol [hunk 5]
```diff
@@ -72,4 +87,13 @@ library LiquidationBidPriorityQueue {
 
         self.liquidationBidsMap[rank] = liquidationBid;
     }
+
+    function hasActiveQueue(LiquidationBidPriorityQueues storage queues) internal view returns (bool) {
+
+        return (
+            queues.latestQueueEndTimestamp != 0
+            && block.timestamp < queues.latestQueueEndTimestamp
+            && queues.priorityQueues[queues.latestQueueId].ranks.length < 0
```
Comment (0xZenus) [inline core/src/libraries/LiquidationBidPriorityQueue.sol:96]: shouldn't be `>0`?

File: core/src/libraries/LiquidationBidPriorityQueue.sol [hunk 6]
```diff
@@ -72,4 +87,13 @@ library LiquidationBidPriorityQueue {
 
         self.liquidationBidsMap[rank] = liquidationBid;
     }
+
+    function hasActiveQueue(LiquidationBidPriorityQueues storage queues) internal view returns (bool) {
```
Comment (0xZenus) [inline core/src/libraries/LiquidationBidPriorityQueue.sol:91]: I'd rename it `hasActiveBids` rather than queue. wdyt?

Comment (avniculae) [reply core/src/libraries/LiquidationBidPriorityQueue.sol]: [e9c1183](https://github.com/Voltz-Protocol/v2-core/pull/228/commits/e9c1183e61a0e72810612eac028a7b23858302e8)

File: core/src/libraries/LiquidationBidPriorityQueue.sol (hunk 5 above)
Comment (avniculae) [reply core/src/libraries/LiquidationBidPriorityQueue.sol]: [e9c1183](https://github.com/Voltz-Protocol/v2-core/pull/228/commits/e9c1183e61a0e72810612eac028a7b23858302e8)

**Review by avniculae — APPROVED** [review]

**Pull request merged**